flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DeploymentHandle.java

```diff
@@ -66,4 +70,14 @@ public DeploymentOption getDeploymentOption() {
 	public SlotExecutionVertexAssignment getSlotExecutionVertexAssignment() {
 		return slotExecutionVertexAssignment;
 	}
+
+	public Optional<LogicalSlot> getLogicalSlot() {
+		final CompletableFuture<LogicalSlot> logicalSlotFuture = slotExecutionVertexAssignment.getLogicalSlotFuture();
+		Preconditions.checkState(logicalSlotFuture.isDone(), "method can only be called after slot future is done");
+
+		if (logicalSlotFuture.isCompletedExceptionally() || logicalSlotFuture.isCancelled()) {
```

Comment:
`#isCancelled()` is not needed since it is one of the `#isCompletedExceptionally()` cases.

Reply:
you are right

Reply:
fixed